File: ISL.ReIdentification.Core/Models/Foundations/OdsDatas/OdsData.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using ISL.ReIdentification.Core.Models.Foundations.PdsDatas;

namespace ISL.ReIdentification.Core.Models.Foundations.OdsDatas
{
    public class OdsData : IKey
    {
        public Guid Id { get; set; }
        public string OrganisationCode_Root { get; set; }
        public string OrganisationPrimaryRole_Root { get; set; }
        public string OrganisationCode_Parent { get; set; }
        public string OrganisationPrimaryRole_Parent { get; set; }
        public DateTimeOffset RelationshipStartDate { get; set; }
        public DateTimeOffset RelationshipEndDate { get; set; }
        public string Path { get; set; }
        public int Depth { get; set; }
        public PdsData PdsData { get; set; }
    }
}

File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Client/src/models/odsData/odsData.ts
import { Guid } from "guid-typescript";
import { PdsData } from "../pds/pdsData";
export class Ods {
    public id: Guid;
    public organisationCode_Root: string;
    public organisationPrimaryRole_Root: string;
    public organisationCode_Parent: string;
    public organisationPrimaryRole_Parent: string;
    public relationshipStartDate?: Date | undefined;
    public relationshipEndDate?: Date | undefined;
    public path?: string;
    public depth?: number;
    public pdsData?: PdsData

    constructor(ods: any) {
        this.id = ods.id ? Guid.parse(ods.id) : Guid.parse(Guid.EMPTY);
        this.organisationCode_Root = ods.organisationCode_Root || "";
        this.organisationPrimaryRole_Root = ods.organisationPrimaryRole_Root || "";
        this.organisationCode_Parent = ods.organisationCode_Parent || "";
        this.organisationPrimaryRole_Parent = ods.organisationPrimaryRole_Parent || "";
        this.relationshipStartDate = ods.relationshipStartDate ? new Date(ods.relationshipStartDate) : undefined;
        this.relationshipEndDate = ods.relationshipEndDate ? new Date(ods.relationshipEndDate) : undefined;
        this.path = ods.path || "";
        this.depth = ods.depth;

        if (ods.pdsData !== undefined && ods.pdsData !== null) {
            this.pdsData = new PdsData(ods.pdsData);
        }
    }
}